### PR TITLE
swev-id: astropy__astropy-14182 ascii.rst: support header_rows for multi-row headers

### PR DESCRIPTION
**Pull request by rowan-stein**
This PR enables header_rows in the ascii.rst writer so that multi-line headers (e.g., name + unit) are supported, analogous to other fixed-width writers.\n\nKey changes\n- RST.__init__ now accepts header_rows and passes it through to FixedWidth.\n- RST.write now determines the rule line from the number of header rows and bookends the table with that rule.\n\nBehavior\n- Default remains unchanged for single header row.\n- With header_rows=["name", "unit"], the output matches simple RST table format: top rule, the two header rows, header/body rule, data rows, bottom rule.\n\nTests\n- Added a test to verify the multi-row header output for name+unit.\n\nCloses #13

### CODE REVIEW

File: astropy/io/ascii/tests/test_rst.py
Comment (noa-lucent) [inline astropy/io/ascii/tests/test_rst.py]: The new test uses `assert_equal_splitlines(...)` but this helper isn’t imported in this file. Please add it to the imports (e.g., `from .common import assert_equal_splitlines`) so the test runs. Without this, the test will raise `NameError`.

Comment (noa-lucent) [inline astropy/io/ascii/tests/test_rst.py]: The expected output asserts exact numeric formatting for the wave column (e.g., `350.0`). Depending on dtype/formatter, quantities with integer inputs can render as `350` (no decimal). To make the test robust, either:
- Construct `wave` with floats (`[350.0, 950.0] * u.nm`), or
- Specify a format for the column (e.g., `tbl['wave'].format = '0.1f'`), or
- Relax the assertion to tolerate either form.

Otherwise this test may be flaky across versions or platform defaults.

File: astropy/io/ascii/rst.py
Comment (noa-lucent) [inline astropy/io/ascii/rst.py]: Backwards compatibility: the previous implementation did `lines = [lines[1]] + lines + [lines[1]]`. With a single header row, your `sep_index` should resolve to 1 to preserve behavior. This hinges on pulling `header_rows` from the correct source (see earlier comment). Please add an explicit unit test for the single-row default to ensure the output is unchanged.

File: astropy/io/ascii/tests/test_rst.py [hunk 1]
```diff
@@ -185,3 +189,26 @@ def test_write_normal():
...
+    """Write a SimpleRST table with multiple header rows (name, unit)."""
+    tbl = QTable({"wave": [350, 950] * u.nm, "response": [0.7, 1.2] * u.count})
+    out = StringIO()
+    ascii.write(tbl, out, Writer=ascii.RST, header_rows=["name", "unit"])
+    # Column widths are computed across header and data rows:
+    # wave column width = max(len('wave'), len('nm'), len('350.0'), len('950.0')) = 5
+    # response column width = max(len('response'), len('ct'), len('0.7'), len('1.2')) = 8
+    # Thus, the rule lines have 5 and 8 '=' characters respectively.
+    assert_equal_splitlines(
+        out.getvalue(),
```
Comment (noa-lucent) [inline astropy/io/ascii/tests/test_rst.py]: Consider making the units consistent with typical RST and Quantity display. `u.count` often renders as `ct`; if the library normalizes to `ct`, the expected string is correct. If a future change normalizes differently, this assertion becomes brittle. One way to guard is to assert the structure (rule rows count and positions), and specific header content tokens (`wave`, `response`, unit substrings) separately from exact spacing where reasonable.

File: astropy/io/ascii/rst.py [hunk 2]
```diff
@@ -57,10 +57,29 @@ class RST(FixedWidth):
...
-        super().__init__(delimiter_pad=None, bookend=False)
+    def __init__(self, header_rows=None):
+        """Initialize the RST writer.
+
+        Parameters
+        ----------
+        header_rows : list of str or None
+            Optional list of column attributes to include as header rows.
+            Examples: ["name"], ["name", "unit"]. When None, defaults to
+            ["name"].
```
Comment (noa-lucent) [inline astropy/io/ascii/rst.py]: Minor: `header_rows : list of str or None` — in NumPy-style docstrings use `list[str] | None`. If you keep `list of str`, ensure consistency with nearby writer docstrings.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Thanks for the clear PR and tests. The approach makes sense and aligns with other fixed-width writers. I’m requesting changes for a couple of issues that affect correctness and test stability:

- Correctness: `sep_index` reads `header_rows` from `self.data`, which likely won’t be set. This can select the wrong line as the rule when multi-row headers are used. Please source `header_rows` from the writer/header or detect the rule line from the emitted lines.
- Back-compat: Add an explicit test confirming single-row default behavior is unchanged (bookend rule equals the header/body separator at index 1).
- Tests: Import `assert_equal_splitlines`, and make the numeric/unit formatting stable (e.g., float inputs or explicit format). Clean up minor style nits in the test for consistency.
- Docs: Small docstring/type tweaks and an optional note on rule detection would improve clarity.

Once these are addressed, I expect this to be ready to approve.

**Comment by rowan-stein** [comment]
Addressed feedback:\n- Rule line now selected using header.header_rows (not data).\n- Added test asserting default single-row header output unchanged.\n- Added multi-row header test using QTable with units.\n- Minor docstring and comments clarified.\n\nPlease re-review.

File: astropy/io/ascii/rst.py [hunk 3]
```diff
@@ -57,10 +57,30 @@ class RST(FixedWidth):
...
+        """
+        # Pass through header_rows to FixedWidth to allow multi-row headers.
+        super().__init__(delimiter_pad=None, bookend=False, header_rows=header_rows)
 
     def write(self, lines):
+        # Collect the core lines from FixedWidth writer which include:
```
Comment (noa-lucent) [inline astropy/io/ascii/rst.py]: Edge case: if `self.header.header_rows` is explicitly `None`, `len(header_rows)` will raise. Consider a safer fallback:

`header_rows = getattr(self.header, 'header_rows', None) or ['name']`

This preserves back-compat even if upstream sets `header_rows=None` to mean default behavior.

**Review by noa-lucent — COMMENTED** [review]
Thanks for the quick updates — the switch to `header.header_rows` and the added tests address the main concerns. I left two small follow-ups (import for `assert_equal_splitlines` and a minor edge-case guard for `header_rows=None`). Once those are fixed, this looks good to approve.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Thanks for the updates — the implementation and added tests address the core concerns. One remaining blocker: the new tests call `assert_equal_splitlines` but it is not imported from `.common`, which will raise a NameError. Please add that import. 

Optional (non-blocking): consider guarding `header_rows=None` with `getattr(self.header, 'header_rows', None) or ['name']` for extra resilience.

Once the import is added, I’m ready to approve.

**Comment by rowan-stein** [comment]
Follow-up commits pushed:\n- Import assert_equal_splitlines in test_rst.py.\n- Guard header_rows retrieval in RST.write for None.\n\nPlease take another look.

File: astropy/io/ascii/tests/test_rst.py
Comment (noa-lucent) [inline astropy/io/ascii/tests/test_rst.py]: Looks good. Minor follow-up for future hardening (non-blocking): to avoid potential float formatting drift across versions, you could set explicit formats, e.g., `tbl['wave'].format = '0.1f'` and `tbl['response'].format = '0.1f'`. Current expectations seem stable, so this is optional.

**Review by noa-lucent — APPROVED** [review]
Changes look good:
- Rule line derived from `header.header_rows` with safe fallback
- Explicit tests for default single-row and multi-row name+unit behaviors
- Docstring/comments clarified

LGTM. Approving.
